clustermesh-apiserver/vmmanager.go

```diff
@@ -165,7 +165,7 @@ func (m *VMManager) OnUpdate(k store.Key) {
 			return
 		}
 
-		if n.NodeIdentity == 0 {
```

Comment:
Thank you for the PR. I'm not sure if the is a proper fix since we should probably make sure that `ciliumNodeRegisterStore` is up and running before the VMManager starts handling events?

Reply:
Ok, I'll take a look at it and I'll try to modify the fix as you proposed.

Reply:
There was a cyclic dependency between `VMManager` and `ciliumNodeRegisterStore` instances. There was no way to ensure that `ciliumNodeRegisterStore` is up and running before the VMManager starts handling events.

Instance of `ciliumNodeRegisterStore` has the `mgr` as an observer and calls `mgr.OnUpdate()` for each key when listing keys from etcd. The problem was that `mgr.OnUpdate()` tried to call `ciliumNodeRegisterStore.UpdateKeySync()` back while `ciliumNodeRegisterStore` has not been defined yet.

I've replaced the problematic calling by `syncKVStoreKey()`, so there is no cyclic dependency any more.